workers.go

```diff
@@ -54,9 +54,13 @@ func NewWorkerManager(id string, config *ConsumerConfig, topicPartition TopicAnd
 	availableWorkers := make(chan *Worker, config.NumWorkers)
 	for i := 0; i < config.NumWorkers; i++ {
 		workers[i] = &Worker{
-			OutputChannel: make(chan WorkerResult),
-			TaskTimeout:   config.WorkerTaskTimeout,
+			InputChannel:         make(chan *TaskAndStrategy),
+			OutputChannel:        make(chan WorkerResult),
+			HandlerInputChannel:  make(chan *TaskAndStrategy),
+			HandlerOutputChannel: make(chan WorkerResult),
+			TaskTimeout:          config.WorkerTaskTimeout,
 		}
+		workers[i].Start()
```

Comment:
lets chat about this. would be great to have some comments about what each of these do too